### DISCUSSION TITLE
Koishi 4.17.10

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#1428] [General]
## Features

- **cordis:** 支持了 `ctx.mixin()` 的副作用回收 (cordiverse/cordis@773a762aea1ca5d6db38faf54489ee55a567e080)
- **create:** 初始化项目时将写入 `packageManager` (cordiverse/cordis@2bdbf0cc5155d12bf6fa8c9cf9303efe5da110bd)
- **discord:** 支持使用 `global_name` 属性作为 `user.nick` (satorijs/satori#285) (satorijs/satori@4a1f4871224657077fd6e93d56c25e5361b3c3f8)
- **discord:** 实验性地支持了 `<sticker>` 元素 (未来将会统一为标准元素) (satorijs/satori#284) (satorijs/satori@8d1a614daaf7eeb0ddad73009d4971cca5c8e368)
- **qq:** 更新了官方 API 的鉴权方式和事件回复 (satorijs/satori#290) (satorijs/satori@4c68dc235aabd5af9b938ce4276b87a35aa0826a)
- **qq:** 支持了获取群成员头像 (satorijs/satori#293) (satorijs/satori@eebde783eee56ed59c9bae429de1b85ed460bdbc)

## Bug Fixes

- **cordis:** 修复了打包器环境下的 `process.cwd()` 调用失败的问题 (cordiverse/cordis@6d7cfe6ad6c3247900d830d6ba785abc7998bd2f)
- **cordis:** 适配了 Node 20.13 以上版本的模块内部 API (cordiverse/cordis@624d27f5e54886bd2514afd3a81ad44c3c70c576)
- **inspect:** 修复了引用回复消息 `messageId` 缺失的问题 (koishijs/koishi#1417) (koishijs/koishi@e4fe81b65e4a963cbd8d3a6c9028543a8af7546f)
- **koishi:** 解析指令参数时支持提取内部资源元素 (koishijs/koishi#1400) (koishijs/koishi@eb7714a057bc713d424f84153b58169eade2c57f)
- **koishi:** 指令选项别名支持使用 CJK 字符了 (koishijs/koishi#1424) (koishijs/koishi@605b1b7fe17ea42d830437bf4e38ced5b6a93092)
- **koishi:** 修复了 `ctx.broadcast()` 永远报错失败的问题 (koishijs/koishi#1418) (koishijs/koishi@10e359a7d2bc2d67360b29896d66f89036220996)
- **qq:** 修复了部分情况下 `<audio>` 的发送 (satorijs/satori#289) (satorijs/satori@3bba22c7d4dbef8d5216ed583b8bb5ce25bf0d6a)
- **qq:** 修复了格式更改后 `session.timestamp` 的解析 (satorijs/satori#296) (satorijs/satori@1080073a63b860e11b2522152dae897a10e3fc36)
- **schema:** 适配了新版 element-plus 的样式变动 (shigma/schemastery@69fe8ef0f03248625b95401c1d8fb0027a24d3e3)

## Other Changes

- chore: bump versions (koishijs/koishi@b2462893f99a78e7e98bd0ef110f253eebf0a337)
- refa: refactor cordis api (cordiverse/http@d9725b1b3270caabcb10e3c1bc3b3a8d4d0bfcbd)
- chore: bump versions (cordiverse/http@83bbf8c74bcd84bf26b69ecc44abc7b28ee6fb9c)
- chore: re-order mixed decl, fix koishijs/webui#331 (koishijs/webui@14ec1b6164cec194b1725f7cd076622e76cb946f)
- refa: use message encoder for mock bots (koishijs/koishi@e97a63106b7d30f59697f75d3c684a2bcec658d9)
- chore: bump versions (cordiverse/cordis@2d4a1f605dd55b3f87e992c178b51728aca1d442)
- chore: bump versions (shigma/schemastery@35d8ef9d60085835b6945c04278f365688b1b7e2)
- chore: fix import and config (cordiverse/webui@01e05a06bf93090267fc8050e768bab5228b39c6)
- chore: bump versions (satorijs/satori@57047bc057648a4497a06efeeca785761cb99180)
- chore: bump versions (koishijs/koishi@7c9d21d2410f5150f77adac3792f8662f3657838)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.17.10'>Koishi 4.17.10</a>.</em>

**Comment by yebei199** [comment]
好, 很有精神
